flake-module.nix

```diff
@@ -10,6 +10,8 @@ let
   inherit (types)
     functionTo
     raw;
+  root-files = builtins.attrNames (builtins.readDir ./.);
+  currentCabalAttrs = builtins.listToAttrs (builtins.map (cabalFile : { name = pkgs.lib.strings.removeSuffix ".cabal" cabalFile; value = builtins.toString ./.;}) (builtins.filter (name: pkgs.lib.strings.hasSuffix ".cabal" name) root-files));
```

Comment:
Instead of `pkgs.lib`, you can also just use `lib` (since it is in the function argument; see line 1).

Also,

- Instead of `./.`, I believe you want to use `self`. Because, the former refers to haskell-flake itself, where as the latter (`self`)` refers to the Haskell project on which this module is being uised.
- `value` should be `{ root = self; }` I think.